packages/core-sdk/src/resources/royalty.ts

```diff
@@ -287,19 +360,38 @@ export class RoyaltyClient {
       });
       await this.rpcClient.waitForTransactionReceipt({ hash });
       txHashes.push(hash);
-
-      // auto unwrap WIP tokens once they are transferred
-      if (token === WIP_TOKEN_ADDRESS && !skipUnwrapIp) {
-        const withdrawalHash = await this.wrappedIpClient.withdraw({
-          value: amount,
-        });
-        txHashes.push(withdrawalHash);
-        await this.rpcClient.waitForTransactionReceipt({ hash: withdrawalHash });
-      }
     };
     for (const { token, amount } of claimedTokens) {
       await transferToken(token, amount);
     }
     return txHashes;
   }
+
+  private async getClaimerInfo(claimer: Address) {
+    const isClaimerIp = await this.ipAssetRegistryClient.isRegistered({
+      id: claimer,
+    });
+    const ipAccount = new IpAccountImplClient(this.rpcClient, this.wallet, claimer);
+    let ownsClaimer = claimer === this.walletAddress;
+    if (isClaimerIp) {
+      const ipOwner = await ipAccount.owner();
+      ownsClaimer = ipOwner === this.walletAddress;
+    }
+    return { ownsClaimer, isClaimerIp, ipAccount };
+  }
+
+  private async unwrapWIPTokens(claimedTokens: IpRoyaltyVaultImplRevenueTokenClaimedEvent[]) {
+    const txHashes: Hex[] = [];
```

Comment:
can we make sure we never have `claimedTokens` with multiple wip records? We should never get there as it should already be aggregated. We then should update this to return return a single txHash.